snap/validate.go

```diff
@@ -47,6 +48,85 @@ func ValidateEpoch(epoch string) error {
 	return nil
 }
 
+// ValidateLicense checks if a string is a valid SPDX expression.
+func ValidateLicense(license string) error {
```

Comment:
I need more direction before making changes to this branch. This is a parser based off the [SPDX Specification 2.1](https://spdx.org/spdx-specification-21-web-version) appendix IV.